torch/_prims/utils.py

```diff
@@ -0,0 +1,468 @@
+import torch
+
+from numbers import Number
+from typing import Union, Sequence
+from functools import reduce
+
+
+class TensorMeta(object):
+    """
+    Temporary helper class to model tensor metadata.
+
+    To be replaced with an actual meta tensor.
+    """
+
+    def __init__(
+        self, tensorlike=None, *, shape=None, strides=None, dtype=None, device=None
+    ):
+
+        if isinstance(tensorlike, Number):
+            self.shape = ()
+            self.strides = ()
+            self.dtype = type_to_dtype(type(tensorlike))
+            self.device = torch.device("cpu")
+        elif tensorlike is not None:
+            self.shape = tensorlike.shape
+            self.strides = tensorlike.stride()
+            self.dtype = tensorlike.dtype
+            self.device = tensorlike.device
+        else:
+            assert shape is not None
+            assert strides is not None
+            assert dtype is not None
+            assert device is not None
+
+        self.shape = self.shape if shape is None else shape
+        self.strides = self.strides if strides is None else strides
+        self.dtype = self.dtype if dtype is None else dtype
+        self.device = self.device if device is None else device
+
+        assert isinstance(self.shape, Sequence)
+        assert isinstance(self.strides, Sequence)
+        assert isinstance(self.dtype, torch.dtype)
+        assert isinstance(self.device, torch.device)
+
+        self.ndim = len(self.shape)
+
+    def stride(self):
+        return self.strides
+
+    def numel(self):
+        if len(self.shape) == 0:
+            return 1
+
+        return reduce(lambda x, acc: x * acc, self.shape, 1)
+
+
+TensorLike = (torch.Tensor, TensorMeta)
+
+
+def compare_tensor_meta(a, b):
+    assert isinstance(a, TensorLike)
+    assert isinstance(b, TensorLike)
+
+    for x, y in zip(a.shape, b.shape):
+        assert x == y
```

Comment:
To make this code serious, will need to stop using `assert` and do a proper test and then throw error when it fails

Reply:
Yeah a message would be nice